src/Middleware/ResponseCaching/src/Streams/SegmentReadStream.cs

```diff
@@ -1,225 +0,0 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
-
-using System;
-using System.Collections.Generic;
-using System.IO;
-using System.Threading;
-using System.Threading.Tasks;
-
-namespace Microsoft.AspNetCore.ResponseCaching
-{
-    internal class SegmentReadStream : Stream
-    {
-        private readonly List<byte[]> _segments;
-        private readonly long _length;
-        private int _segmentIndex;
-        private int _segmentOffset;
-        private long _position;
-
-        internal SegmentReadStream(List<byte[]> segments, long length)
-        {
-            _segments = segments ?? throw new ArgumentNullException(nameof(segments));
-            _length = length;
-        }
-
-        public override bool CanRead => true;
-
-        public override bool CanSeek => true;
-
-        public override bool CanWrite => false;
-
-        public override long Length => _length;
-
-        public override long Position
-        {
-            get
-            {
-                return _position;
-            }
-            set
-            {
-                // The stream only supports a full rewind. This will need an update if random access becomes a required feature.
-                if (value != 0)
-                {
-                    throw new ArgumentOutOfRangeException(nameof(value), value, $"{nameof(Position)} can only be set to 0.");
-                }
-
-                _position = 0;
-                _segmentOffset = 0;
-                _segmentIndex = 0;
-            }
-        }
-
-        public override void Flush()
-        {
-            throw new NotSupportedException("The stream does not support writing.");
-        }
-
-        public override int Read(byte[] buffer, int offset, int count)
-        {
-            if (buffer == null)
```

Comment:
Was read ever used? Otherwise, makes sense.

Reply:
`SegmentReadStream` was used to wrap the data stored in the cache, so `Read` was using when serving from cache.
Now I'm using CachedResponseBody just as a wrapper of the byte segments, which then copy to the pipewriter without using a stream, so this is no longer needed.
Is that what you were asking?